tests/FileStatus.hs

```diff
@@ -63,10 +65,35 @@ testSymlink fs ds = do
   when (statusElements ds /= statusElements ds') $
     fail "status for a directory does not match when it's accessed via a symlink"
 
+
+testLink = do
+  createLink regular hlink_regular
+  (fs, _)  <- getStatus regular -- we need to retrieve it again as creating the link causes it to change!
+  (fs', ls)  <- getStatus hlink_regular
+  let expected = (
+                False, -- isBlockDevice
+                False, -- isCharacterDevice
+                False, -- isNamedPipe
+                True,  -- isRegularFile
+                False, -- isDirectory
+                False, -- isSymbolicLink
+                False) -- isSocket
+      actualF  = snd (statusElements ls)
+
+  when (actualF /= expected) $
+    fail "unexpected file status bits for hard link to regular file"
+
+  when (linkCount fs' /= 2) $
+    fail "newly created hard link was expected to contain have a link count of 2"
+
+  when (statusElements fs /= statusElements fs') $
+    fail "status for a file does not match when it's accessed via a link"
```

Comment:
Since you expanded this test suite, I think it's high time to improve its UX. Let's use [`assertEqual`](http://hackage.haskell.org/package/tasty-hunit-0.10.0.3/docs/Test-Tasty-HUnit.html#v:assertEqual) or [`@?=`](http://hackage.haskell.org/package/tasty-hunit-0.10.0.3/docs/Test-Tasty-HUnit.html#v:-64--61--63-) to report what exactly mismatches. You can find an example in [`ResourceLimit`](https://github.com/haskell/unix/blob/cdff32fd57cc5a1e6d7f1c08a9e4ea390284d9c8/tests/ResourceLimit.hs) test.

Reply:
Done for the newer test. I think it's outside of the scope of this PR to improve the other tests, but it can serve as an example